gradio/routes.py

```diff
@@ -376,14 +364,17 @@ async def conditional_routing_middleware(request: Request, call_next):
                     and path not in ["/config", "/login"]
                     and not path.startswith("/theme")
                 ):
-                    print("proxying")
-                    try:
+                    if App.app_port is None:
+                        App.app_port = request.url.port or int(
+                            os.getenv("GRADIO_SERVER_PORT", "7860")
+                        )
 
+                    try:
                         return await App.proxy_to_node(
                             request,
-                            request.client.host,
+                            os.getenv("GRADIO_SERVER_NAME") or request.client.host,
```

Comment:
SHould we set a different default here? `0.0.0.0` is probably safer that client.host anytime someone is proxying tbh.